### ISSUE TITLE
Allow users to share a document with other users at the same or lower role

### ISSUE THREAD

**Issue by mitar** [peer/doc#146]
So currently only admins can invite (or even see permissions). I think it would be better to allow users to invite other users at the role they have or lower. (Internally, it would be in fact allow users to invite other users and give them permissions they have or a subset of them.)

Also, everyone should be able to see all permissions on the document, if they have access to the document. I think this is important for accountability and transparency.

**Comment by subjective** [comment]
I agree that everyone should always be able to see permissions. I can't think of any use case in which this would be undesirable.

In principle, I like the idea of allowing users to invite others at the same role or lower. The only problem is that this change would create a potential vulnerability for voting. A user can invite multiple friends or fake accounts to accumulate voting power. To be sure, the problem is mitigated by making all permissions transparent.

But the vulnerability would complicate one of the main use cases for PeerDoc: a group with a clearly defined set of members (e.g., co-op, union, university) needs to produce a document, and a trusted member (admin) simply creates a document and shares it with all members, each of whom has the same voting power. If we allow users to mess with the distribution of voting power, there will be a lot of confusion. We should keep this use case simple.

Perhaps we should have a binary configuration for admins, determining whether users can invite others.

**Comment by mitar** [comment]
Hm, I think you are mixing voting question with permissions to edit the document. I am not sure why editing of the document should not be simply delegated further if wanted.

I can see the issue with commenting, that it could happen that it leads to a form of astroturfing where one person would invite many fake bots to comment as well, so a variation of what you are describing. But I think we should generally have protections against that in the app.

I would keep voting separate from these permissions. And I could also see that we could in the future use voting delegation to address the issue you are mentioning. Admin would select from users who can vote and then you can delegate that permission further, but it works that if you do not vote, then the other's vote is used as yours, but if you do vote, then this is it and other user (to whom you delegated) cannot vote. Or we even do not allow that to be delegated. But I think voting permission is simply a separate permission which might need special handling, or special ways of computing votes.

**Comment by subjective** [comment]
Good. I agree. Then let's just keep voting as a special kind of permission, and allow any user A to give any other user B the same or lower level of regular permissions (viewing, editing, commenting) that A has.

**Comment by mitar** [comment]
Thanks for pointing out that we should be careful about how the voting permissions follow such delegation.
